pandas/tests/frame/test_nonunique_indexes.py

```diff
@@ -513,3 +513,31 @@ def test_set_value_by_index(self):
 
         df.iloc[:, 0] = 3
         tm.assert_series_equal(df.iloc[:, 1], expected)
+
+
+def test_iloc_with_boolean_operation():
+    # GH 20627
+    result = DataFrame([[0, 1], [2, 3], [4, 5], [6, np.nan]])
+    result.iloc[result.index <= 2] *= 2
```

Comment:
you need to copy result *each* time you are using it

Reply:
Are you sure ? `result` is modified *inplace* ? (or I don't understand)

Reply:
i get it. for each part of the test you want to reconstruct result, so that the in place mods only affect 1 part and are not consecutive.